apps/api/src/app/events/e2e/digest-events.e2e.ts

```diff
@@ -812,14 +826,110 @@ describe('Trigger event - Digest triggered events - /v1/events/trigger (POST)',
       _templateId: template._id,
       _subscriberId: subscriber._id,
     });
-
-    expect(messages[0].content).to.include(digests[0].payload.postId);
     expect(messages[1].content).to.include(digests[1].payload.postId);
+    expect(messages[2].content).to.include(digests[2].payload.postId);
+    const jobCount = await jobRepository.count({
+      _environmentId: session.environment._id,
+      _templateId: template._id,
+    });
+    const allJobs = await jobRepository.find({
+      _environmentId: session.environment._id,
+      _templateId: template._id,
+    });
+    expect(jobCount).to.equal(11);
+  });
+
+  it('should create multiple digests based on different nested digestKeys', async function () {
+    const postId = MessageRepository.createObjectId();
+    const postId2 = MessageRepository.createObjectId();
+
+    template = await session.createTemplate({
+      steps: [
+        {
+          type: StepTypeEnum.DIGEST,
+          content: '',
+          metadata: {
+            unit: DigestUnitEnum.MINUTES,
+            amount: 5000,
+            digestKey: 'nested.postId',
+            type: DigestTypeEnum.REGULAR,
+          },
+        },
+        {
+          type: StepTypeEnum.CHAT,
+          content: 'Hello world {{nested.postId}}' as string,
+        },
+      ],
+    });
+    await triggerEvent({
+      customVar: 'No digest key',
+    });
+
+    await triggerEvent({
+      customVar: 'digest key1',
+      nested: { postId: postId },
+    });
+    await triggerEvent({
+      customVar: 'digest key2',
+      nested: { postId: postId2 },
+    });
+    await session.awaitParsingEvents();
+
+    await triggerEvent({
+      customVar: 'No digest key repeat',
+    });
+    await triggerEvent({
+      customVar: 'digest key1 repeat',
+      nested: { postId: postId },
+    });
+    await session.awaitParsingEvents();
+
+    await session.awaitRunningJobs(template?._id, false, 3);
+
+    let digests = await jobRepository.find({
+      _environmentId: session.environment._id,
+      _templateId: template._id,
+      type: StepTypeEnum.DIGEST,
+    });
+    expect(digests[1].payload.nested.postId).not.to.equal(digests[2].payload.nested.postId);
+    expect(digests[0].payload.nested?.postId).to.equal(undefined);
+
+    expect(digests.length).to.equal(3);
```

Comment:
Makes more sense this check to be done before the ones of the nested postId as if this one is not accomplished the others won't and checking it later is not totally useful (I always expect tests to fail as early as possible).
Can you change the order in this test and the other ones where the lenght of the digests are checked after doing other array checks?